packages/components/src/components/NFT/NFT.tsx

```diff
@@ -15,24 +22,66 @@ export interface NFTProps {
  * TODO: Add a component that fetchs the NFT data from the blockchain and uses this component to
  * display it
  */
-export const NFT = ({
-  tokenId,
-  name,
-  imageUrl,
-  assetContractName,
-  assetContractSymbol,
-}: NFTProps) => {
-  const displayName = name || tokenId;
+export const NFT = ({ contractAddress, tokenId }: NFTProps) => {
+  const _isMounted = useRef(true);
+  const [nftData, setNftData] = React.useState<NFTData>();
+
+  const fetchNFTData = useCallback(async () => {
+    const response = await (
+      await fetch(`https://api.opensea.io/api/v1/asset/${contractAddress}/${tokenId}/`)
+    ).json();
+    if (_isMounted.current) {
+      setNftData({
+        tokenId: response.token_id,
+        imageUrl: response.image_url,
+        name: response.name,
+        assetContractName: response.asset_contract.name,
+        assetContractSymbol: response.asset_contract.symbol,
+        animationUrl: response.animation_url,
+      });
+    }
+  }, [contractAddress, tokenId]);
+
+  useEffect(() => {
+    _isMounted.current = true;
+    fetchNFTData();
+    return () => {
+      _isMounted.current = false;
+    };
+  }, [contractAddress, tokenId]);
+
+  if (!nftData) {
+    return null;
```

Comment:
maybe let's render a loader with a box around it the size of the NFT card by default, so that we don't have the cards pop in and move the layout around?

Reply:
Yep, good idea